### PR TITLE
Depot List & Create Depot

### PR DESCRIPTION
**Pull request by SMenigat**
#### Summary
Adds depot management, which allows to list existing depot and create new ones.

#### Description
- fixes apiWatcher saga to run in while(true)
- fixes snackbar to close itself permanently
- enhances Scrollable with footer & floatingActionButton props
- enhances apiWacher saga to support callback
- adds DepotList container
- adds DepotCreate container
- adds DepotForm component

#### Internationalization
- [x] no hardcoded texts
- [x] translations are implemented

#### Tests

- ~~Utils~~
- Redux
    - [x] Action creators
    - [x] Reducers
    - ~~Sagas~~
    - [x] Selectors
- Components
    - [x] Rendered elements (also different states)
    - [x] Callbacks (e.g. event handlers)
    - [x] Helper functions that are specific to the component
    - [x] Life cycle hooks (e.g. component will receive props)

### CODE REVIEW

File: app/containers/DepotCreate/actions.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      DEPOT_CREATE_FAILURE,
+    ],
+    config: {
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      method: 'POST',
+      body: JSON.stringify({
+        id: 0,
```
Comment (PhilippSpo) [inline app/containers/DepotCreate/actions.js]: Do we need to pass `id: 0` here? Does this make sense? 🤔

File: app/containers/DepotCreate/index.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                onTouchTap={() => {
+                  this.depotForm.submit();
+                }}
+              />
+            </ToolbarGroup>
+          </Toolbar>
+        )}
+      >
+        <DepotForm
+          handleSubmit={this.handleSubmit}
```
Comment (PhilippSpo) [inline app/containers/DepotCreate/index.js]: Do we really need `handleSubmit` and `onSubmit`?

File: app/containers/DepotCreate/actions.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { CALL_API } from '../App/constants';
+import {
+  DEPOT_CREATE_REQUEST,
+  DEPOT_CREATE_SUCCESS,
+  DEPOT_CREATE_FAILURE,
+} from './constants';
+import schemas from 'schemas';
+
+
+export const createDepot = (name, address, zip, city, callback) => ({
```
Comment (PhilippSpo) [inline app/containers/DepotCreate/actions.js]: With so many params, it is hard to get the order right. We could follow the pattern, that (async) action creators always get 2 params - data and callback:
```js
const createDepot = ({ name, address, zip, city }, callback) => ({
```

File: app/containers/DepotCreate/tests/index.test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import expect from 'expect';
+import { shallow } from 'enzyme';
+import React from 'react';
+import sinon from 'sinon';
+
+import { DepotCreate } from '../index';
+
+const getProps = () => ({
```
Comment (PhilippSpo) [inline app/containers/DepotCreate/tests/index.test.js]:
```js
const createProps = (custom) => ({
  // props here
  ...custom
})
```
And later you can make your code easier:
```js
const props = createProps({
  createDepot: (name, address, zip, city, callback) => callback(),
});
```

File: app/containers/DepotList/index.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            secondary
+          >
+            <FontIcon className="material-icons">add</FontIcon>
+          </FloatingActionButton>
+        )}
+      >
+        {this.props.isFetching ? (
+          <CircularProgress size={60} thickness={7} />
+        ) : (
+          <div>
```
Comment (PhilippSpo) [inline app/containers/DepotList/index.js]: Do we really need the `<div>` around the `<List>`?

File: app/containers/DepotList/tests/index.test.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
+import expect from 'expect';
+import { shallow } from 'enzyme';
+import React from 'react';
+import sinon from 'sinon';
+
+import { DepotList } from '../index';
+
+const getProps = () => ({
```
Comment (PhilippSpo) [inline app/containers/DepotList/tests/index.test.js]: Again, we can accept custom props here 😁

**Comment by PhilippSpo** [comment]
Funzt auch super. N bisschen Padding um die Form wär noch cool gewesen 😉

**Comment by SMenigat** [comment]
Jo, wollte jetzt aber noch nix stylen. Erstmal nur Funktionalität 😎

**Pull request merged**